cluster.c

```diff
@@ -38,15 +227,31 @@ static int addClusterSlotNodeReply(RedisRaftCtx *rr, RedisModuleCtx *ctx, raft_n
     RedisModule_ReplyWithCString(ctx, addr->host);
     RedisModule_ReplyWithLongLong(ctx, addr->port);
 
-    snprintf(node_id, sizeof(node_id) - 1, "%040x", raft_node_get_id(raft_node));
+    snprintf(node_id, sizeof(node_id) - 1, "%.32s%08x", rr->log->dbid, raft_node_get_id(raft_node));
     RedisModule_ReplyWithCString(ctx, node_id);
 
     return 1;
 }
 
+static int addClusterSlotShardGroupNodeReply(RedisRaftCtx *rr, RedisModuleCtx *ctx, ShardGroupNode *sgn)
+{
+    /* Create a three-element reply:
+     * 1) Address
+     * 2) Port
+     * 3) Node ID
+     */
+
+    RedisModule_ReplyWithArray(ctx, 3);
+    RedisModule_ReplyWithCString(ctx, sgn->addr.host);
+    RedisModule_ReplyWithLongLong(ctx, sgn->addr.port);
+    RedisModule_ReplyWithCString(ctx, sgn->node_id);
+
+    return 1;
+}
+
 static void addClusterSlotsReply(RedisRaftCtx *rr, RaftReq *req)
 {
-    int i;
+    int i, j;
```

Comment:
It would be cleaner to declare these vars when they are first used, inside the `for`.